src/dom/Struct.ts

```diff
@@ -46,8 +46,11 @@ export class Struct extends Value(
     annotations: string[] = []
   ) {
     super();
+    // Struct fields will always be stored as (fieldName, fieldValues) => (string, Value[])
     for (const [fieldName, fieldValue] of fields) {
-      this._fields[fieldName] = fieldValue;
+      this._fields[fieldName] = Array.isArray(fieldValue)
```

Comment:
`dom.List` and `dom.SExpression` inherit from `Array` -- is this able to distinguish between an `Array` and a `Value` that happens to inherit from `Array`?

Reply:
No. Its doesn't distinguish. Maybe I can reverse the check and see if it's an `instanceof Value` first and otherwise it will always be `Value[]` array because of the constructor arguments.